src/rust/runtime/queue/qresult.rs

```diff
@@ -7,16 +7,33 @@
 
 use crate::runtime::{
     fail::Fail,
+    memory::DemiBuffer,
     network::types::Port16,
     QDesc,
 };
-use ::std::net::Ipv4Addr;
+use ::std::{
+    fmt,
+    net::{
+        Ipv4Addr,
+        SocketAddrV4,
+    },
+};
 
 //==============================================================================
-// Enumerations
+// Structures
 //==============================================================================
 
+pub enum OperationResult {
+    Connect,
+    Accept((QDesc, SocketAddrV4)),
+    Push,
+    // TODO: Drop wrapping Option.
```

Comment:
The option is required because the underlying pop may fail to return the address of the remote.

Reply:
This comment was there from before. I can take it out however.

Reply:
Oh, sure! Could you remove it?

Reply:
Yup, done